Framework/Core/src/TableTreeHelpers.cxx

```diff
@@ -85,310 +85,124 @@ class ColumnIterator
-        mVariable_us += mNumberElements;
-        mValueBuffer = (void*)mVariable_us;
-        break;
-      case arrow::Type::type::UINT32:
-        mVariable_ui += mNumberElements;
-        mValueBuffer = (void*)mVariable_ui;
-        break;
-      case arrow::Type::type::UINT64:
-        mVariable_ul += mNumberElements;
-        mValueBuffer = (void*)mVariable_ul;
-        break;
-      case arrow::Type::type::INT8:
-        mVariable_b += mNumberElements;
-        mValueBuffer = (void*)mVariable_b;
-        break;
-      case arrow::Type::type::INT16:
-        mVariable_s += mNumberElements;
-        mValueBuffer = (void*)mVariable_s;
-        break;
-      case arrow::Type::type::INT32:
-        mVariable_i += mNumberElements;
-        mValueBuffer = (void*)mVariable_i;
-        break;
-      case arrow::Type::type::INT64:
-        mVariable_l += mNumberElements;
-        mValueBuffer = (void*)mVariable_l;
-        break;
-      case arrow::Type::type::FLOAT:
-        mVariable_f += mNumberElements;
-        mValueBuffer = (void*)mVariable_f;
-        break;
-      case arrow::Type::type::DOUBLE:
-        mVariable_d += mNumberElements;
-        mValueBuffer = (void*)mVariable_d;
-        break;
-      default:
-        LOGP(FATAL, "Type {} not handled!", mElementType);
-        break;
-    }
-  }
-  mBranchPtr->SetAddress(mValueBuffer);
-
-  return true;
-}
-
-TableToTree::TableToTree(std::shared_ptr<arrow::Table> table,
-                         TFile* file,
-                         const char* treename)
-{
-  mTable = table;
-
-  // try to get the tree
-  mTreePtr = (TTree*)file->Get(treename);
-
-  // create the tree if it does not exist already
-  if (!mTreePtr) {
-    // does treename containe folder?
-    std::string treeName(treename);
-    auto pos = treeName.find_first_of("/");
-    if (pos != std::string::npos) {
-      file->cd(treeName.substr(0, pos).c_str());
-    }
-    treeName = treeName.substr(pos + 1, std::string::npos);
-    mTreePtr = new TTree(treeName.c_str(), treeName.c_str());
-  }
-}
-
-bool TableToTree::addBranch(std::shared_ptr<arrow::ChunkedArray> col, std::shared_ptr<arrow::Field> field)
-{
-  auto brit = std::make_unique<BranchIterator>(mTreePtr, col, field);
-  if (!brit->getStatus()) {
-    return false;
-  }
-  mBranchIterators.emplace_back(std::move(brit));
-  return true;
-}
-
-bool TableToTree::addAllBranches()
-{
-
-  bool status = mTable->num_columns() > 0;
-
-  for (auto ii = 0; ii < mTable->num_columns(); ii++) {
-    status &= addBranch(mTable->column(ii), mTable->schema()->field(ii));
+      throw runtime_error_f("Unsupported column type for %s", field->name().c_str());
   }
-
-  return status;
 }
 
 TTree* TableToTree::process()
 {
-
-  bool togo = (mTreePtr->GetNbranches() > 0) && (mTable->num_rows() > 0);
-  while (togo) {
-    // fill the tree
-    mTreePtr->Fill();
-
-    // update the branches
-    for (auto& brit : mBranchIterators) {
-      togo &= brit->push();
+  int64_t mRow = 0;
```

Comment:
do not prefix local variables with `m`.

Reply:
Leftover from this variable being a data member, I will rename it.

Reply:
Fixed.